src/shogun/clustering/KMeansMiniBatch.cpp

```diff
@@ -71,18 +72,7 @@ void KMeansMiniBatch::minibatch_KMeans()
 			SGVector<float64_t> dists=SGVector<float64_t>(k);
 			for (int32_t p=0; p<k; p++)
 				dists[p]=distance->distance(M[j],p);
-
-			int32_t imin=0;
-			float64_t min=dists[0];
-			for (int32_t p=1; p<k; p++)
-			{
-				if (dists[p]<min)
-				{
-					imin=p;
-					min=dists[p];
-				}
-			}
-			ncent[j]=imin;
+			ncent[j] = Math::arg_max(dists.vector, 1, dists.vlen);
```

Comment:
I thought this was meant to find argmin?

Reply:
oh, yes, I will change it to argmin, thanks you point it.

Reply:
huh if this has not been caught by CI means that we dont have a test that implicitly tests stuff of this part of the code.... @LiuYuHui might be good to add some unit test that covers this :)

Reply:
plz avoid testing private functions of KMeansMiniBatch.... just public interface

Reply:
ok, I am very happy to do it!

Reply:
great thnx! just add it to this PR

Reply:
hey @vigsterkr , does it look fine to you?